packages/nextjs/src/utils/config/webpack.ts

```diff
@@ -0,0 +1,206 @@
+  WebpackEntryProperty,
+} from './types';
+import {
+  SENTRY_CLIENT_CONFIG_FILE,
+  SENTRY_SERVER_CONFIG_FILE,
+  SERVER_SDK_INIT_PATH,
+  storeServerConfigFileLocation,
+} from './utils';
+
+export { SentryWebpackPlugin };
+
+// TODO: merge default SentryWebpackPlugin ignore with their SentryWebpackPlugin ignore or ignoreFile
+// TODO: merge default SentryWebpackPlugin include with their SentryWebpackPlugin include
+// TODO: drop merged keys from override check? `includeDefaults` option?
+
+const defaultSentryWebpackPluginOptions = dropUndefinedKeys({
+  url: process.env.SENTRY_URL,
+  org: process.env.SENTRY_ORG,
+  project: process.env.SENTRY_PROJECT,
+  authToken: process.env.SENTRY_AUTH_TOKEN,
+  configFile: 'sentry.properties',
+  stripPrefix: ['webpack://_N_E/'],
+  urlPrefix: `~/_next`,
+  include: '.next/',
+  ignore: ['.next/cache', 'server/ssr-module-cache.js', 'static/*/_ssgManifest.js', 'static/*/_buildManifest.js'],
+});
+
+/**
+ * Construct the function which will be used as the nextjs config's `webpack` value.
+ *
+ * Sets:
+ *   - `devtool`, to ensure high-quality sourcemaps are generated
+ *   - `entry`, to include user's sentry config files (where `Sentry.init` is called) in the build
+ *   - `plugins`, to add SentryWebpackPlugin (TODO: optional)
+ *
+ * @param userNextConfig The user's existing nextjs config, as passed to `withSentryConfig`
+ * @param userSentryWebpackPluginOptions The user's SentryWebpackPlugin config, as passed to `withSentryConfig`
+ * @returns The function to set as the nextjs config's `webpack` value
+ */
+export function constructWebpackConfigFunction(
+  userNextConfig: ExportedNextConfig = {},
+  userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
+): WebpackConfigFunction {
+  const newWebpackFunction = (config: WebpackConfigObject, options: BuildContext): WebpackConfigObject => {
+    // if we're building server code, store the webpack output path as an env variable, so we know where to look for the
+    // webpack-processed version of `sentry.server.config.js` when we need it
+    if (config.target === 'node') {
+      storeServerConfigFileLocation(config);
+    }
+
+    let newConfig = config;
+
+    // if user has custom webpack config (which always takes the form of a function), run it so we have actual values to
+    // work with
+    if ('webpack' in userNextConfig && typeof userNextConfig.webpack === 'function') {
+      newConfig = userNextConfig.webpack(config, options);
+    }
+
+    // Ensure quality source maps in production. (Source maps aren't uploaded in dev, and besides, Next doesn't let you
+    // change this is dev even if you want to - see
+    // https://github.com/vercel/next.js/blob/master/errors/improper-devtool.md.)
+    if (!options.dev) {
+      // TODO Handle possibility that user is using `SourceMapDevToolPlugin` (see
+      // https://webpack.js.org/plugins/source-map-dev-tool-plugin/)
+      // TODO Give user option to use `hidden-source-map` ?
```

Comment:
lets make sure each of these todos have tickets - we can then discuss with the team and anyone can pick them up.

Maybe even open a GH issue and make it open for contribution? Could be a good way to get the community involved.